pkg/apis/pipeline/v1/param_types.go

```diff
@@ -314,12 +314,24 @@ func validatePipelineParametersVariablesInTaskParameters(params []Param, prefix
 	return errs
 }
 
-// validatePipelineParametersVariablesInMatrixParameters validates matrix param value
+// validatePipelineParametersVariablesInMatrixParameters validates all pipeline paramater variables including Matrix.Params and Matrix.Include.Params
 // that may contain the reference(s) to other params to make sure those references are used appropriately.
-func validatePipelineParametersVariablesInMatrixParameters(matrix []Param, prefix string, paramNames sets.String, arrayParamNames sets.String, objectParamNameKeys map[string][]string) (errs *apis.FieldError) {
-	for _, param := range matrix {
-		for idx, arrayElement := range param.Value.ArrayVal {
-			errs = errs.Also(validateArrayVariable(arrayElement, prefix, paramNames, arrayParamNames, objectParamNameKeys).ViaFieldIndex("value", idx).ViaFieldKey("matrix", param.Name))
+func validatePipelineParametersVariablesInMatrixParameters(matrix *Matrix, prefix string, paramNames sets.String, arrayParamNames sets.String, objectParamNameKeys map[string][]string) (errs *apis.FieldError) {
+	if matrix != nil {
+		if matrix.MatrixHasInclude() {
+			for _, include := range matrix.Include {
+				for idx, param := range include.Params {
+					stringElement := param.Value.StringVal
+					errs = errs.Also(validateStringVariable(stringElement, prefix, paramNames, arrayParamNames, objectParamNameKeys).ViaFieldIndex("value", idx).ViaFieldKey("matrix.include.params", param.Name))
+				}
+			}
+		}
```

Comment:
please add tests for these changes

Reply:
I believe the tests aded in this PR: [TEP-0118: Add validation for matrix pipeline context parameter variables](https://github.com/tektoncd/pipeline/pull/6238/files#top) should cover this. Just waiting for a /lgtm and then can merge the changes.

Reply:
ok, will put a hold on this until the other PR is merged

/hold

Reply:
The tests have been added. 

/hold cancel

Reply:
/unhold